### PR TITLE
typstwriter: init at 0.1

### PR DESCRIPTION
**Pull request by drupol**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ty/typstwriter/package.nix [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  src = fetchFromGitHub {
+    owner = "Bzero";
+    repo = "typstwriter";
+    rev = "V${version}";
+    hash = "sha256-xgBBZTViMzYgxaYb24druUwLqVWdf9utCETC+goLqYk=";
+  };
+
+  nativeBuildInputs = [
+    python3.pkgs.flit-core
```
Comment (GaetanLepage) [inline pkgs/by-name/ty/typstwriter/package.nix:20]: This might belong to `build-system`.

File: pkgs/by-name/ty/typstwriter/package.nix [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      pytest-qt
+    ];
+  };
+
+  pythonImportsCheck = [ "typstwriter" ];
+
+  meta = {
+    description = "An integrated editor for the typst typesetting system";
+    homepage = "https://github.com/Bzero/typstwriter";
+    license = lib.licenses.mit;
```
Comment (GaetanLepage) [inline pkgs/by-name/ty/typstwriter/package.nix:46]:
```suggestion
    homepage = "https://github.com/Bzero/typstwriter";
    changelog = "https://github.com/Bzero/typstwriter/releases/tag/V${version}";
    license = lib.licenses.mit;
```

**Review by GaetanLepage — DISMISSED** [review]

File: pkgs/by-name/ty/typstwriter/package.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      pytest
+      pytest-qt
+    ];
+  };
+
+  pythonImportsCheck = [ "typstwriter" ];
+
+  meta = {
+    changelog = "https://github.com/Bzero/typstwriter/releases/tag/V${version}";
+    description = "An integrated editor for the typst typesetting system";
```
Comment (Aleksanaa) [inline pkgs/by-name/ty/typstwriter/package.nix:40]:
```suggestion
    description = "Integrated editor for the typst typesetting system";
```

Guess what? Our contribution guidelines has changed! https://github.com/NixOS/nixpkgs/tree/master/pkgs#meta-attributes

File: pkgs/by-name/ty/typstwriter/package.nix [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+python3.pkgs.buildPythonApplication rec {
+  pname = "typstwriter";
+  version = "0.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "Bzero";
+    repo = "typstwriter";
+    rev = "V{version}";
```
Comment (Aleksanaa) [inline pkgs/by-name/ty/typstwriter/package.nix:15]:
```suggestion
    rev = "V${version}";
```

(Huh)

Comment (drupol) [reply pkgs/by-name/ty/typstwriter/package.nix:15]: oops, brainfart.

**Comment by drupol** [comment]
@ofborg build typstwriter

**Comment by drupol** [comment]
@ofborg build typstwriter

**Pull request merged**